deployment-template/eks/otel-fargate-container-insights.yaml

```diff
@@ -0,0 +1,552 @@
+            operations:
+              - action: update_label
+                label: namespace
+                new_label: Namespace
+              - action: update_label
+                label: pod
+                new_label: PodName
+          - include: ^new_container_(.*)$$
+            match_type: regexp
+            action: update
+            new_name: container_$$1
+
+      # add cluster name from env variable and EKS metadata
+      resourcedetection:
+        detectors: [env, eks]
+
+      batch:
+        timeout: 60s
+
+    exporters:
+      awsemf:
+        log_group_name: '/aws/containerinsights/{ClusterName}/performance'
+        log_stream_name: '{PodName}'
+        namespace: 'ContainerInsights'
+        region: us-east-1
+        resource_to_telemetry_conversion:
+          enabled: true
+        eks_fargate_container_insights_enabled: true
+        parse_json_encoded_attr_values: ["kubernetes"]
+        dimension_rollup_option: NoDimensionRollup
+        metric_declarations:
+          - dimensions: [ [ClusterName, LaunchType], [ClusterName, Namespace, LaunchType], [ClusterName, Namespace, PodName, LaunchType]]
+            metric_name_selectors:
+              - pod_cpu_utilization_over_pod_limit
+              - pod_cpu_usage_total
+              - pod_cpu_limit
+              - pod_memory_utilization_over_pod_limit
+              - pod_memory_working_set
+              - pod_memory_limit
+              - pod_network_rx_bytes
+              - pod_network_tx_bytes
+
+      logging:
+        loglevel: debug
+
+    extensions:
+      health_check:
+
+    service:
+      pipelines:
+        metrics:
+          receivers: [prometheus]
+          processors: [metricstransform/label_1, resourcedetection, metricstransform/rename, filter, cumulativetodelta, deltatorate, experimental_metricsgeneration/1, experimental_metricsgeneration/2, metricstransform/label_2, batch]
+          exporters: [logging, awsemf]
+      extensions: [health_check]
+
+# configure the service and the collector as a StatefulSet
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: adot-collector-service
+  namespace: fargate-container-insights
+  labels:
+    app: aws-adot
+    component: adot-collector
+spec:
+  ports:
+    - name: metrics # default endpoint for querying metrics.
+      port: 8888
+  selector:
+    component: adot-collector
+  type: ClusterIP
+
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: adot-collector
+  namespace: fargate-container-insights
+  labels:
+    app: aws-adot
+    component: adot-collector
+spec:
+  selector:
+    matchLabels:
+      app: aws-adot
+      component: adot-collector
+  serviceName: adot-collector-service
+  template:
+    metadata:
+      labels:
+        app: aws-adot
+        component: adot-collector
+    spec:
+      serviceAccountName: adot-collector
+      securityContext:
+        fsGroup: 65534
+      containers:
+        - image: amazon/aws-otel-collector:latest
```

Comment:
Needs a specific ADOT Collector version tag.

Reply:
Let's use the latest version `v0.15.1`. Please test in house as well. Though we already tested with :latest tag which is the same version I believe.